quasar-core/src/main/java/co/paralleluniverse/strands/channels/TransferChannel.java

```diff
@@ -414,10 +417,10 @@ void won() {
             try {
                 UNSAFE = UtilUnsafe.getUnsafe();
                 Class k = Node.class;
-                itemOffset = UNSAFE.objectFieldOffset(k.getDeclaredField("item"));
-                saOffset = UNSAFE.objectFieldOffset(k.getDeclaredField("sa"));
-                nextOffset = UNSAFE.objectFieldOffset(k.getDeclaredField("next"));
-                waiterOffset = UNSAFE.objectFieldOffset(k.getDeclaredField("waiter"));
+                itemOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(k, "item")));
+                saOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(k, "sa")));
+                nextOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(k, "next")));
+                waiterOffset = UNSAFE.objectFieldOffset(doPrivileged(new GetDeclaredField(k, "waiter")));
             } catch (Exception e) {
                 throw new Error(e);
```

Comment:
Shouldn't you unwrap the the `PriviledgedAccessException` here as cause for the error?

Reply:
Same in the following few files